numpy/core/src/private/mem_overlap.c

```diff
@@ -415,7 +415,7 @@ diophantine_dfs(unsigned int n,
             x[0] = x1 + c1*t_l;
             x[1] = x2 - c2*t_l;
             if (require_ub_nontrivial) {
-                int j, is_ub_trivial;
+                unsigned int j, is_ub_trivial;
```

Comment:
Using `unsigned` for `is_ub_trivial` is a little odd - would prefer to leave it as `int`, or switch to `npy_bool`

Reply:
Yes, it only takes the values 0 and 1. However, see above about major refactoring of the code.

Reply:
This particular one seems like a step in the wrong directions, without the justification of fixing any warnings.

But I suppose I'm being overly picky - as a whole, the patch is better than doing nothing.

Reply:
OK, I made `is_ub_trivial` an integer.

Reply:
It's not that I disagree with you, it's that in my past experience with doing number theoretic stuff with fixed sized integers lots of things can go wrong, so I regard this code as likely somewhat delicate. Enough so, that I wouldn't want to make extensive changes without thoroughly understanding the algorithm, which is why I pinged @pv.